src/libre/parser.act

```diff
@@ -1080,7 +1090,7 @@
 		act_state->overlap = overlap;
 		act_state->poolp   = &ast->pool;
 
-		act_state->group_id = 1;
+		act_state->group_id = 0;
```

Comment:
There's a comment around line 127 near the top of this file about the numbering for `group_id`; if we're numbering from 0 now, I think that needs updating. (Or I don't follow what the 0 here means)

Reply:
It says 
```
		/*
		 * Numbering for capturing groups. By convention these start from 1,
		 * and 0 represents the entire matching text.
		 */
```
which is still accurate, we are just changing how the capture group 0 for the entire matching text is represented so that we can keep a prefix/suffix match for an unanchored start/end outside of group 0. Previously we started counting from 1, because group 0 was implicit.

Reply:
Hm I don't think I understand. It sounds like the `.group_id` field can now never be zero, in which case I think it'd be helpful to explain in the comment that group 0 is represented elsewhere.

Reply:
Does this sound clearer?

> By convention group 0 implicitly represents the entire matching text (not including any prefix/suffix ignored due to an unanchored start/end), and groups 1 and on represent capture groups in the regular expression.

I would prefer to only change the phrasing once, because each change will also involve rebuilding the parsers (which contain the comment, but otherwise not change).

Reply:
That makes sense, and it's true for regexps in general. But what's eating at me here is I still don't understand the possible values for the `.group_id` struct member. Can that ever be 0? That's what I'm after for the comment.